test/src/main/java/org/apache/accumulo/test/FateSummaryIT.java

```diff
@@ -174,93 +187,32 @@ public void testFatePrintAndSummaryCommandsWithInProgressTxns() throws Exception
-        }
-
-        // Keep printing until we see a transaction complete mid-print or until we run out of
-        // transactions (they all complete).
-        // Realistically, should only take 1 or 2 iterations to see a transaction complete
-        // mid-print.
-        do {
-          // Execute the command when transactions are currently running and may complete mid-print
-          p = execAdminFateCommand(command);
-          // Previously, this check could fail due to a ZK NoNodeException
-          assertEquals(0, p.getProcess().waitFor());
-          result = p.readStdOut();
-          // A transaction should have completed mid-print and been ignored
-        } while (!result.contains("Tried to get info on a since completed transaction - ignoring")
-            && !noTransactions(result, command));
-
-        if (noTransactions(result, command)) {
-          // Fail since we printed until all transactions have completed and didn't see a
-          // transaction complete mid-print.
-          // This is highly unlikely to have occurred.
-          fail();
-        }
-        // Otherwise, we saw 'Tried to get info on a since completed transaction - ignoring', so
-        // test passes
-        client.tableOperations().delete(table);
-      }
-    }
-  }
-
-  private boolean noTransactions(String result, String command) {
-    if (command.equals("--print")) {
-      return result.contains(" 0 transactions");
-    } else { // --summary
-      result = result.substring(result.indexOf("{"), result.lastIndexOf("}") + 1);
-      FateSummaryReport report = FateSummaryReport.fromJson(result);
-      return report != null && report.getReportTime() != 0 && report.getStatusCounts().isEmpty()
-          && report.getStepCounts().isEmpty() && report.getCmdCounts().isEmpty()
-          && report.getStatusFilterNames().isEmpty() && report.getFateDetails().isEmpty();
-    }
-  }
-
-  private ProcessInfo execAdminFateCommand(String command) throws Exception {
-    if (command.equals("--print")) {
-      return getCluster().exec(Admin.class, "fate", command);
-    } else { // --summary
-      return getCluster().exec(Admin.class, "fate", command, "-j");
+    ServerContext sctx = getCluster().getServerContext();
+
+    // This error was occurring in AdminUtil.getTransactionStatus(). One of the methods that is
+    // called which may throw the NNE is top(), so we will mock this method to always throw a NNE
+    // and ensure it is handled/ignored within getTransactionStatus()
+    ZooStore<String> zs = EasyMock.createMockBuilder(ZooStore.class)
+        .withConstructor(String.class, ZooReaderWriter.class)
+        .withArgs(sctx.getZooKeeperRoot() + Constants.ZFATE, sctx.getZooReaderWriter())
+        .addMockedMethod("top").createMock();
+    expect(zs.top(anyLong())).andThrow(new RuntimeException(new KeeperException.NoNodeException()))
+        .atLeastOnce();
+    replay(zs);
+    zs.create();
+
```

Comment:
Would it be possible to simulate the following situation in this test?

 1. ZooStore list returns three fate txs
 2. Trying to get information about the 2nd one throws NNE
 3. AdminUtil.getTransactionStatus successfully returns data about the other 2

Not sure how much of a pain this would be to do w/ mocking.  Thinking this verifies not only that the method does not fail when NNE is seen but that it also still returns data for the fate txs that did not throw NNE

Reply:
Yes, I can make that change. That's a good idea. Don't think it will be much of a hassle